packages/gatsby-plugin-feed/src/gatsby-node.js

```diff
@@ -19,9 +28,40 @@ const serialize = ({ query: { site, allMarkdownRemark } }) =>
     }
   })
 
-exports.onPostBuild = async ({ graphql }, pluginOptions) => {
+exports.onPreBootstrap = async function onPreBootstrap(
+  { reporter },
+  pluginOptions
+) {
   delete pluginOptions.plugins
 
+  try {
+    const normalized = await pluginOptionsSchema.validate(pluginOptions)
+
+    // TODO: remove these checks in the next major release
+    if (!normalized.feeds) {
+      reporter.warn(
+        reporter.stripIndent(
+          warnMessage(`feeds option`, `the internal RSS feed creation`)
+        )
+      )
+    } else if (
+      normalized.feeds.some(feed => typeof feed.serialize !== `function`)
+    ) {
+      reporter.warn(
+        reporter.stripIndent(
+          warnMessage(
+            `serialize function in a feed`,
+            `the internal serialize function`
+          )
+        )
+      )
+    }
+  } catch (e) {
+    throw new Error(e.details.map(detail => detail.message).join(`\n`))
```

Comment:
We should mention that this error comes from config validation. I think gatsby will say it comes from `gatsby-plugin-feed`, but figuring out might be confusing without more context

Reply:
Just prefix the error? e.g. `[Config Validation]: ${detail.message}` or something?

Reply:
I imagine you'll want a format that can be (eventually) adopted by other plugins. For now, I agree it makes sense to prefix with `gatsby-plugin-feed`.

Reply:
Sounds good, just something that user isn't greeted with `Error: "query" is required` message and not sure where to go from there